src/Provider/DataSourceProvider.js

```diff
@@ -27,36 +27,48 @@ const error = (err, url, source) => {
     source.handlingError(err, url);
     throw err;
 };
-function FetchAndConvertSourceData(url, layer, extentSource, extentDestination) {
+
+function convertSourceData(data, extDest, layer) {
+    return layer.convert(data, extDest, layer);
+}
+
+function parseSourceData(data, extSrc, extDest, layer) {
     const source = layer.source;
-    const fetcher = source.fetcher || supportedFetchers.get(source.format) || Fetcher.texture;
-    const parser = source.parser || supportedParsers.get(source.format) || (data => data);
+    const parser = source.parser || supportedParsers.get(source.format) || (d => Promise.resolve(d));
 
-    const parsingOptions = {
-        buildExtent: !layer.isGeometryLayer,
+    const options = {
+        buildExtent: source.isFileSource || !layer.isGeometryLayer,
         crsIn: source.projection,
         crsOut: layer.projection,
         // TODO FIXME: error in filtering vector tile
         // filteringExtent: extentDestination.as(layer.projection),
-        filteringExtent: layer.isGeometryLayer ? extentDestination : undefined,
+        filteringExtent: !source.isFileSource && layer.isGeometryLayer ? extDest : undefined,
         overrideAltitudeInToZero: layer.overrideAltitudeInToZero,
         filter: layer.filter,
         isInverted: source.isInverted,
-        mergeFeatures: layer.mergeFeatures === undefined ? true : layer.mergeFeatures,
+        mergeFeatures: source.isFileSource || layer.mergeFeatures === undefined ? true : layer.mergeFeatures,
```

Comment:
the source shouldn't decide in the options parsing, it's rather the layer that influences the options.

Reply:
Ok

Reply:
Removed